packages/enzyme-test-suite/test/RSTTraversal-spec.jsx

```diff
@@ -183,6 +183,36 @@ describe('RSTTraversal', () => {
       expect(spy.callCount).to.equal(4);
     });
 
+    it('should handle array children', () => {
+      const spy = sinon.spy();
+      const twoDivArray = [
+        <div key="a" />,
+        <div key="b" />,
+      ];
+      const node = $(
+        <div>
+          {twoDivArray}
+        </div>,
+      );
+      treeForEach(node, spy);
+      expect(spy.callCount).to.equal(3);
+    });
+
+    it('should handle non-array iterable children', () => {
+      const spy = sinon.spy();
+      const twoDivIterable = new Set([
+        <div key="a" />,
+        <div key="b" />,
+      ]);
+      const node = $(
+        <div>
+          {twoDivIterable}
+        </div>,
+      );
+      treeForEach(node, spy);
+      expect(spy.callCount).to.equal(3);
```

Comment:
Could we also use `expect(spy.calls).to.deep.equal` in both places, and assert that it's called with the right items?

It'd also be great to test with a Map as well as a Set

Reply:
Yes for sure, I like that. The tree objects are pretty complex, how do you feel about just checking part of it, say keys? Like this:

```js
it('should handle Map children', () => {
  const spy = sinon.spy();
  const twoDivMap = new Map([
    [<div key="a" />],
    [<div key="b" />],
  ]);
  const node = $(
    <div key="root">
      {twoDivMap}
    </div>,
  );
  treeForEach(node, spy);
  expect(spy.callCount).to.equal(3);
  const keys = spy.args.map(arg => arg[0].key);
  expect(keys).to.deep.equal(['root', 'a', 'b']);
});

```

Reply:
ah nevermind, even better I can just use `$` and check the whole thing:
```js
it('should handle Map children', () => {
  const spy = sinon.spy();
  const twoDivMap = new Map([
    [<div key="a" />],
    [<div key="b" />],
  ]);
  const divA = $(<div key="a" />);
  const divB = $(<div key="b" />);
  const node = $(
    <div key="root">
      {twoDivMap}
    </div>,
  );
  treeForEach(node, spy);
  expect(spy.callCount).to.equal(3);
  const nodes = spy.args.map(arg => arg[0]);
  expect(nodes).to.deep.equal([node, divA, divB]);
});
```

I'll get this committed soon!